### PR TITLE
Bump Gradle and AGP samples

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
_#skip-changelog_

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
depends on https://github.com/getsentry/sentry-react-native/pull/1836

**Review by romtsn — APPROVED** [review]

**Comment by marandaneto** [comment]
> Configure project :react-native-reanimated
Evaluating project ':react-native-reanimated' using build file '/Users/user/Github/sentry-react-native/sample/node_modules/react-native-reanimated/android/build.gradle'.

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Build file '/Users/user/Github/sentry-react-native/sample/node_modules/react-native-reanimated/android/build.gradle' line: 21

* What went wrong:
A problem occurred evaluating project ':react-native-reanimated'.
> Plugin with id 'maven' not found.

File: sample/android/build.gradle [hunk 1]
```diff
@@ -2,20 +2,16 @@
...
-
-        ndkVersion = "21.4.7075529"
     }
     repositories {
         google()
         mavenCentral()
     }
     dependencies {
-        classpath("com.android.tools.build:gradle:4.2.2")
+        classpath("com.android.tools.build:gradle:7.1.1")
```
Comment (marandaneto) [inline sample/android/build.gradle:14]:
```suggestion
        classpath("com.android.tools.build:gradle:7.1.2")
```

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -59,7 +59,7 @@ jobs:
         with:
           api-level: 29
           # All the tests need to be run in this script as the emulator is killed after this command.
-          script: cd sample && react-native run-android --variant=release && curl --output /dev/null --silent --head --fail http://127.0.0.1:4723/wd/hub/sessions && yarn test
+          script: cd sample && react-native run-android && curl --output /dev/null --silent --head --fail http://127.0.0.1:4723/wd/hub/sessions && yarn test
```
Comment (marandaneto) [inline .github/workflows/e2e.yml:62]: https://github.com/react-native-community/upgrade-support/issues/38#issuecomment-1059532701

can't merge if this does not get fixed

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-react-native/pull/2280

**Pull request closed without merging**